src/types/templates.ts

```diff
@@ -1,12 +1,31 @@
-import type { BindEventForHits } from '../lib/utils';
+import type { VNode } from 'preact';
+import type {
+  Highlight,
+  ReverseHighlight,
+  ReverseSnippet,
+  Snippet,
+} from '../helpers/components';
+import type { html } from 'htm/preact';
+import type { BindEventForHits, SendEventForHits } from '../lib/utils';
 
 export type Template<TTemplateData = void> =
   | string
-  | ((data: TTemplateData) => string);
+  | ((data: TTemplateData, params: TemplateParams) => VNode | VNode[] | string);
+
+export type TemplateParams = BindEventForHits & {
+  html: typeof html;
+  components: {
+    Highlight: typeof Highlight;
+    ReverseHighlight: typeof ReverseHighlight;
+    Snippet: typeof Snippet;
+    ReverseSnippet: typeof ReverseSnippet;
+  };
+  sendEvent?: SendEventForHits;
```

Comment:
should maybe be changed later only, but sendEvent can be for facets too I think (unless we don't expose that?)

Reply:
For now we don't pass either `bindEvent` or `sendEvent` to any other component than `<Hits>` and `<InfiniteHits>`. I agree this can be useful for other widgets though, but we can do it separately.